braidpool-primitives/src/braid/mod.rs

```diff
@@ -42,20 +45,41 @@ impl Braid {
         Braid {
             beads: genesis_beads.clone(),
             tips: genesis_beads.clone(),
-            cohorts: vec![Cohort(genesis_beads)],
+            cohorts: vec![Cohort(genesis_beads.clone())],
             orphan_beads: Vec::new(),
             loaded_beads_in_memory: HashMap::new(),
+            genesis_beads: genesis_beads,
+        }
+    }
+
+    pub fn load_beads_in_memory(&mut self, beads: Vec<Bead>) {
```

Comment:
I think we should put Braid loading/saving in the Node instead of the Braid.

Let's try to keep the Bead and Braid data structures as "simple" data structures, with the Node doing most of the manipulation of them.